sonic-net/sonic-buildimage
# Ensure that media_settings.json exists on Mellanox platforms only when the Host CMIS Management feature is enabled

#### Why I did it
Today, in **Mellanox platforms**, we have a feature called 'host CMIS management'. When this feature is enabled, it means that ASIC and module side configurations will be done by software (SW) instead of by firmware (FW). Currently, ASIC SI parameters are sent to SAI even if the host CMIS management feature is disabled. This PR aims to prevent this situation and ensure that ASIC SI parameters will be posted to the APP DB only if the feature is enabled.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
On Mellanox platforms only, there will be a file called 'media_settings_src.json' present. This file will be located in the same directory as the existing 'media_settings.json' file. It will serve as the source of truth for the ASIC SI parameters. Once the host CMIS management feature is enabled (via sai.profile), the media_settings_src.json will be copied into media_settings.json. Conversely, once this feature is disabled, the media_settings.json file will be deleted. This approach ensures that ASIC SI parameters will be posted to the APP DB only if the feature is enabled, without modifying the generic code.

#### How to verify it
On Mellanox platforms, verify that the media_settings.json file is present in the platform folder after enabling the host CMIS management feature. Additionally, confirm that once the feature is disabled, the media_settings.json file is removed.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
